File: graffiti/src/main/java/com/ue/graffiti/model/TransformPelStep.java
package com.ue.graffiti.model;

import android.graphics.Matrix;
import android.graphics.Region;

import java.util.List;

/**
 * Created by hawk on 2018/1/19.
 */

public class TransformPelStep extends Step {
    //变换前的matrix
    public Matrix toUndoMatrix;
    public Region clipRegion;
    public Pel savedPel;

    public TransformPelStep(List<Pel> pelList, Region clipRegion, Pel pel) {
        super(pelList, pel);
        toUndoMatrix = new Matrix();
        savedPel = curPel.clone();
        this.clipRegion = clipRegion;
    }

    public void setToUndoMatrix(Matrix matrix) {
        toUndoMatrix.set(matrix);
    }
}

File: graffiti/src/main/java/com/ue/graffiti/model/TransformPelStep.kt
package com.ue.graffiti.model

import android.graphics.Matrix
import android.graphics.Region

/**
 * Created by hawk on 2018/1/19.
 */

class TransformPelStep(pelList: MutableList<Pel>, var clipRegion: Region, pel: Pel) : Step(pelList, pel) {
    //变换前的matrix
    var toUndoMatrix: Matrix = Matrix()
    var savedPel: Pel = curPel!!.clone()

    fun setUndoMatrix(matrix: Matrix) {
        toUndoMatrix.set(matrix)
    }
}
